### PR TITLE
Improve docs for optimal_edit_paths

### LINKED ISSUE #3818: I have a question about `nx.optimal_edit_paths(G1, G2)`.
I have been conducting a experiment for computing graph similarity.  
I have a question about `nx.optimal_edit_paths(G1, G2)`.
In the [documentation of it](https://networkx.github.io/documentation/stable/reference/algorithms/generated/networkx.algorithms.similarity.optimal_edit_paths.html), it is written that this function **Returns all minimum-cost edit paths transforming G1 to G2**. 

I use the function in my code and I run it. my code is below. 
Even though `G1` and `G2` are exactly same graph and their graph_edit_cost is 0, 
there a a lot of optimal_edit_paths. 

```python
N = 3
G1 = nx.complete_graph(N)
G2 = nx.complete_graph(N)

edit_path, graph_edit_cost = nx.optimal_edit_paths(G1, G2)
print(f"graph edit cost: {graph_edit_cost}")
for p in edit_path:
    node_edit, edge_edit = p
    print(f"node_edit: {node_edit}")
    print(f"edge_edit: {edge_edit}")
    print("--"*30)
```

the output is below. 
As I said it before, graph_edit_cost is zero because `G1` and `G2` are exactly same graphs. 
but there are lots of optimal path when paths was derived by `nx.optimal_edit_paths(G1, G2)`. 
I don't know why this happend. It doesn't look like 'optimal' because they don't have same edit cost. 

```
graph edit cost: 0.0
node_edit: [(0, 0), (1, 1), (2, 2)]
edge_edit: [((0, 1), (0, 1)), ((0, 2), (0, 2)), ((1, 2), (1, 2))]
------------------------------------------------------------
node_edit: [(0, 0), (2, 1), (1, 2)]
edge_edit: [((0, 2), (0, 1)), ((0, 1), (0, 2)), ((1, 2), (1, 2))]
------------------------------------------------------------
node_edit: [(1, 0), (0, 1), (2, 2)]
edge_edit: [((0, 1), (0, 1)), ((0, 2), (1, 2)), ((1, 2), (0, 2))]
------------------------------------------------------------
node_edit: [(1, 0), (2, 1), (0, 2)]
edge_edit: [((1, 2), (0, 1)), ((0, 1), (0, 2)), ((0, 2), (1, 2))]
------------------------------------------------------------
node_edit: [(2, 0), (0, 1), (1, 2)]
edge_edit: [((0, 2), (0, 1)), ((0, 1), (1, 2)), ((1, 2), (0, 2))]
------------------------------------------------------------
node_edit: [(2, 0), (1, 1), (0, 2)]
edge_edit: [((1, 2), (0, 1)), ((0, 1), (1, 2)), ((0, 2), (0, 2))]
------------------------------------------------------------
```

And also, if the function is correct, How can I transform `G1` to `G2` base on that operations(node_edit, edge_edit)? I don't know how those operations have to be applied to `G2` to make `G2`. 

I always appreciate your help in this library. 
becasue of that, I've learned a lot about network science. 

If you feel my text is rude or not polite, it is because of my lack of english. 
sorry about that.

### PR DESCRIPTION
**Pull request by akshayamadhuri**
fixes #3818

### CODE REVIEW

File: networkx/algorithms/similarity.py [hunk 1]
```diff
@@ -322,7 +322,8 @@ def optimal_edit_paths(
         edge_edit_path : list of tuples ((u1, v1), (u2, v2))
 
     cost : numeric
-        Optimal edit path cost (graph edit distance).
+        Optimal edit path cost (graph edit distance).When the cost
```
Comment (dschult) [inline networkx/algorithms/similarity.py:325]:
```suggestion
        Optimal edit path cost (graph edit distance). When the cost
```

File: networkx/algorithms/similarity.py [hunk 2]
```diff
@@ -334,6 +335,14 @@ def optimal_edit_paths(
     >>> cost
     5.0
 
+    Note
+    -------
+    When the cost is zero, it indicates that G1 and G2 are isomorphic,
+    and the optimal paths represent different isomorphic mappings.
+    To transform G1 into a graph isomorphic to G2, apply the
+    node and edge edits in the optimal paths. In the case of isomorphic
+    graphs, these edits involve renaming nodes and edges to match the structure of G2.
```
Comment (dschult) [inline networkx/algorithms/similarity.py:344]: Is this paragraph only talking about the case when cost is zero? Or are the later sentences applicable more generally?

More to the point:
Can we say anything about what the cost means when it is nonzero?
And do the edit_paths mean anything different when the cost is nonzero?

**Review by dschult — COMMENTED** [review]
Thanks for this!!  It's been a long time coming.

Are you getting this info from the paper, or from the comments on the previous issue?
See also below questions about whether we can say anything about what non-zero cost will mean.

File: networkx/algorithms/similarity.py (hunk 2 above)
Comment (akshayamadhuri) [reply networkx/algorithms/similarity.py:344]: @dschult , the `optimal_edit_paths` function currently focuses on finding minimum-cost edit paths, even for isomorphic graphs. In the case of isomorphic graphs, the cost is zero, and the paths represent isomorphism mappings. For nonzero costs, the function likely identifies minimum-cost transformations, but the documentation could be enhanced to provide clearer details on the meaning of costs and edit paths in such scenarios.

Comment (dschult) [reply networkx/algorithms/similarity.py:344]: > the optimal_edit_paths function currently focuses on finding minimum-cost edit paths, even for isomorphic graphs. In the case of isomorphic graphs, the cost is zero, and the paths represent isomorphism mappings. For nonzero costs, the function likely identifies minimum-cost transformations, but the documentation could be enhanced to provide clearer details on the meaning of costs and edit paths in such scenarios.

This is a good summary. Would you like to include it (or something close to it) in the doc_string, or would you like me to?

**Review by dschult — APPROVED** [review]
I went ahead an updated the Notes section with language similar to what @akshayamadhuri put in the comment. 

I approve this PR

**Review by rossbar — APPROVED** [review]
Thanks @akshayamadhuri and @dschult , I went ahead and pushed up a few formatting changes myself, but this LGTM!

**Pull request merged**